lib/types.ts

```diff
@@ -560,3 +588,42 @@ function checkValidValue(lowerRange: number | bigint, upperRange: number | bigin
     throw 'invalid value';
   }
 }
+
+function toPrimitive_TIME(time: TIME): bigint | number {
+  console.log('time', time);
+  const { value, unit, isAdjustedToUTC } = time;
+
+  const timeValue = typeof value === 'string' ? BigInt(value) : BigInt(value);
+
+  if (isAdjustedToUTC) {
+    return unit === 'MILLIS' ? Number(timeValue) : timeValue;
+  } else {
+    switch (unit) {
+      case 'MILLIS':
+        return Number(adjustToLocalTimestamp(timeValue, { MILLIS: true }));
+      case 'MICROS':
+        return adjustToLocalTimestamp(timeValue, { MICROS: true });
+      case 'NANOS':
+        return adjustToLocalTimestamp(timeValue, { NANOS: true });
+      default:
+        throw new Error(`Unsupported time unit: ${unit}`);
+    }
+  }
+}
+
+function adjustToLocalTimestamp(
+  timestamp: bigint,
+  unit: { MILLIS?: boolean; MICROS?: boolean; NANOS?: boolean }
+): bigint {
+  const localOffset = BigInt(new Date().getTimezoneOffset()) * 60n * 1000n; // Offset in milliseconds
+
+  if (unit.MILLIS) {
+    return timestamp - localOffset;
+  } else if (unit.MICROS) {
+    return timestamp - localOffset * 1000n;
+  } else if (unit.NANOS) {
+    return timestamp - localOffset * 1000000n;
+  }
+
+  throw new Error('Unsupported time unit');
+}
```

Comment:
Helper functions to handle time especially following the spec to map units to specific primitive types